asv_bench/benchmarks/frame_methods.py

```diff
@@ -597,6 +597,19 @@ def time_frame_quantile(self, axis):
         self.df.quantile([0.1, 0.5], axis=axis)
 
 
+class Rank:
+    param_names = ["dtype"]
+    params = [
```

Comment:
IUUC you're switching to a masked algo for everything in which case this should (?) work for nullable numeric dtypes. 

If that's right then we want to include `Int`, `UInt`, `Float` here (and add tests for those - on a quick look I didn't see any)

Reply:
Sorry, I should've explained more clearly in the description. This PR implements `group_rank` (the cython rank function used in  `GroupBy.rank`) using `algos.rank_1d ` (so cuts down on 1 rank implementation). To do this, `algos.rank_1d` takes optional labels from a groupby operation. So the new `labels` argument is not a mask, it's for groupby support.

Hopefully deduplication like this makes future enhancements easier, but for now none of these changes add functionality to `Series.rank` or `DataFrame.rank`. The only enhancement is that `GroupBy.rank` gains object support.